### PR TITLE
Update documentation and minor bug fix

### PR DESCRIPTION
**Pull request by allez-allez-allez**
* Adding more documentation about the module
* Fix unsupported argument for the DynamoDB resource. `point-in-time-recovery` needs to be a block type instead of argument.

### CODE REVIEW

File: _docs/MAIN.md [hunk 1]
```diff
@@ -1,3 +1,63 @@
...
+
+This module, by default, deploys a Watchmaker Linux AutoScaling Group with a desired of 2 instances running on `STIG-Partitioned Enterprise Linux (SPEL) Centos 7` Amazon Machine Image (AMI) distributed by [Plus3 IT Systems](https://github.com/plus3it). Spel is a project that helps create and publish Enterprise Linux images that are partitioned according to the [DISA STIG](http://iase.disa.mil/stigs/os/unix-linux/Pages/red-hat.asp). To learn more about SPEL visit our project space [here](https://github.com/plus3it/spel).
+
+
+* For more information on using `terraform-aws-watchmaker`, go to <https://registry.terraform.io/modules/plus3it/watchmaker>.
+* For more information on installing and using `watchmaker`, go to <https://watchmaker.readthedocs.io>.
+
+## How To Use This Module
+
+This repo is structured as follows:
```
[the remark this answers is not in the document]
Comment (allez-allez-allez) [reply _docs/MAIN.md]: Make sense. Changing `How To Use This Module` to `How To Navigate This Module`

File: README.md [hunk 2]
```diff
@@ -1,6 +1,185 @@
...
 
-Terraform module that installs and configures Hashicorp Vault cluster with HA DyanamoDb storage backend.
+# Vault AWS Terraform Module 
+
+## `terraform-aws-vault`
+
+
+## Overview
+
+This Terraform module installs and configures the Hashicorp Vault cluster with HA DyanamoDb storage backend. This module is built on top of the `terraform-aws-watchmaker` module. You can quickly deploy a single instance of Vault or an auto-scale group of Vault instances.
```
Comment (userhas404d) [inline README.md]:
```suggestion
This Terraform module installs and configures the Hashicorp Vault cluster with HA DyanamoDb storage backend. This module is built on top of the `terraform-aws-watchmaker` module. You can quickly deploy a single instance of Vault or an auto-scaled group of Vault instances.
```

File: README.md [hunk 3]
```diff
@@ -1,6 +1,185 @@
...
+2. Create `pillar` folder and files in the following structure.
+```
+├── pillar
+│   ├── top.sls
+│   └── vault
+│       └── init.sls
+``` 
+> **NOTE**: See [`tests\vault-py2\pillar`](tests\vault-py2\pillar) for an example on how to structure the `pillar` folder.
+	 
+3. The run the following command to start executing the module:
```
Comment (userhas404d) [inline README.md]:
```suggestion
3. Then run the following command to deploy the module:
```

File: README.md [hunk 4]
```diff
@@ -1,6 +1,185 @@
...
+              # List auth methods
+              'sys/auth': {capabilities: [read]}
+              # List existing secret engines.'
+              sys/mounts': {capabilities: [read]}
+              # Read health check
+              'sys/health': {capabilities: [read, sudo]}
+```
+> ***Note***: Additional configurations can be specified for authentication methods using the `auth_config` pillar item. This also applies for secrets engines. Specifying the configuration for a particular secrets engine under the `secret_config` pillar item of that secrets engine type.
+
+In some use cases, passwords or sensitive information will need to be provided in order for Vault to communicate and function properly. For instance, when enabling the [Active Directory Secrets Engine](https://www.vaultproject.io/api/secret/ad/index.html), you need to specify the url of the LDAP server, a bind_dn and a bind_pass to perform user search. This information will need to be hidden from the public and only available to `salt` when it synchronizes the configs. One way to specify these config is via the input  `vault_pillar_extra_config` variable within the `terraform.auto.tfvar` file.
```
Comment (userhas404d) [inline README.md]: is this integration limited to LDAPS?

Comment (allez-allez-allez) [reply README.md]: This implementation supports all types of secrets engines mentioned [here](https://www.vaultproject.io/api/secret/index.html). This doc is calling out this specific use case of AD. `LDAPS` works as well as `LDAP`.

**Pull request merged**